src/KokkosComm/channel.hpp

```diff
@@ -0,0 +1,102 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2025) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#pragma once
+
+#include <Kokkos_Core.hpp>
+
+#include "KokkosComm/fwd.hpp"
+//#include "commmode.hpp"
+//#include "impl/pack_traits.hpp"
+//#include "impl/include_mpi.hpp"
+
+namespace KokkosComm {
+
+template <class CommSpace>
+class Channel {
+ public:
+  using comm_space = CommSpace;
+
+  explicit Channel(int num_reqs = 2)  // Example: 2 requests for send/recv
+      : state_(0), send_parts_(0), recv_parts_(0), ready_count_(0), num_reqs_(num_reqs), requests_(num_reqs_) {}
+
+  explicit Channel(int dest_rank, int src_rank, int tag, MPI_Comm comm, int num_reqs = 2)
+      : state_(0),
+        send_parts_(0),
+        recv_parts_(0),
+        ready_count_(0),
+        num_reqs_(num_reqs),
+        requests_(num_reqs_),
+        dest_rank_(dest_rank),
+        src_rank_(src_rank),
+        tag_(tag),
+        comm_(comm) {}
+
+  ~Channel() {
+    // delete[] requests_;  // TODO: Only free if in inactive state
+  }
+
+  template <class SendView>
+  void sendinit(SendView view) {
+    Kokkos::Tools::pushRegion("KokkosComm::Channel::sendinit");
+    using value_type = typename SendView::value_type;
+    // Initialize persistent send
+    MPI_Send_init(view.data(), view.size(), KokkosComm::Impl::mpi_type_v<value_type>, dest_rank_, tag_, comm_,
+                  &(requests_[0].mpi_request()));
```

Comment:
I think it's better to use our own traits here:
```suggestion
    MPI_Send_init(KokkosComm::data_handle(view), KokkosComm::span(view), KokkosComm::Impl::mpi_type_v<value_type>,
                  dest_rank_, tag_, comm_, &(request_[0].mpi_request()));
```

Reply:
Great point! Also resolved in [3711203](https://github.com/kokkos/kokkos-comm/pull/156/commits/37112038658c9db23112f5ab63915ec4807d5cd4)